ako-gateway-api/nodes/dequeue_ingestion.go

```diff
@@ -279,28 +281,83 @@ func (o *AviObjectGraph) ProcessRouteDeletion(key, parentNsName string, routeMod
 
 	parentNode := o.GetAviEvhVS()
 	routeTypeNsName := routeModel.GetType() + "/" + routeModel.GetNamespace() + "/" + routeModel.GetName()
+	if akogatewayapilib.IsGatewayInDedicatedMode(routeModel.GetNamespace()) {
+		o.ProcessRouteDeletionForDedicatedMode(key, parentNsName, routeModel, fullsync)
 
-	found, childVSNames := akogatewayapiobjects.GatewayApiLister().GetRouteToChildVS(routeTypeNsName)
-
-	if found {
-		utils.AviLog.Infof("key: %s, msg: child VSes retrieved for deletion %v", key, childVSNames)
-
-		for _, childVSName := range childVSNames {
-			removed := nodes.RemoveEvhInModel(childVSName, parentNode, key)
-			if removed {
-				akogatewayapiobjects.GatewayApiLister().DeleteRouteChildVSMappings(routeTypeNsName, childVSName)
+	} else {
+		found, childVSNames := akogatewayapiobjects.GatewayApiLister().GetRouteToChildVS(routeTypeNsName)
+		if found {
+			utils.AviLog.Infof("key: %s, msg: child VSes retrieved for deletion %v", key, childVSNames)
+
+			for _, childVSName := range childVSNames {
+				removed := nodes.RemoveEvhInModel(childVSName, parentNode, key)
+				if removed {
+					akogatewayapiobjects.GatewayApiLister().DeleteRouteChildVSMappings(routeTypeNsName, childVSName)
+				}
 			}
 		}
 	}
-
 	updateHostname(key, parentNsName, parentNode[0])
 	modelName := parentNode[0].Tenant + "/" + parentNode[0].Name
 	ok := saveAviModel(modelName, o.AviObjectGraph, key)
 	if ok && len(o.AviObjectGraph.GetOrderedNodes()) != 0 && !fullsync {
 		sharedQueue := utils.SharedWorkQueue().GetQueueByName(utils.GraphLayer)
 		nodes.PublishKeyToRestLayer(modelName, key, sharedQueue)
 	}
+}
+
+func (o *AviObjectGraph) ProcessRouteDeletionForDedicatedMode(key, parentNsName string, routeModel RouteModel, fullsync bool) {
+	utils.AviLog.Infof("key: %s, msg: Processing route deletion for dedicated mode: %s/%s", key, routeModel.GetNamespace(), routeModel.GetName())
+
+	gatewayVSes := o.GetAviEvhVS()
+	if len(gatewayVSes) == 0 {
+		utils.AviLog.Errorf("key: %s, msg: No Gateway VS found for dedicated mode deletion", key)
+		return
+	}
+
+	dedicatedVS := gatewayVSes[0]
+
+	httpPSName := akogatewayapilib.GetHttpPolicySetName(dedicatedVS.AviMarkers.GatewayNamespace, dedicatedVS.AviMarkers.GatewayName, routeModel.GetNamespace(), routeModel.GetName())
+
+	var updatedHttpPolicyRefs []*nodes.AviHttpPolicySetNode
+	for _, policy := range dedicatedVS.HttpPolicyRefs {
+		if policy.Name != httpPSName {
+			updatedHttpPolicyRefs = append(updatedHttpPolicyRefs, policy)
```

Comment:
Since we are only supporting one gateway->  one route configuration will the control ever come to line 325